WcaOnRails/app/javascript/auto-numeric/index.js

```diff
@@ -0,0 +1,67 @@
+import AutoNumeric from 'autonumeric';
+import currenciesData from 'wca/currenciesData.js.erb';
+
+let autoNumericObjectMap = new Map();
+
+function getCurrencyInfo(isoCode) {
+  return currenciesData.byIso[isoCode] || currenciesData.byIso.USD;
+}
+
+// Create a mask for an amount input in the given currency, using autoNumeric
+// 'action' can be either "init" or "update"
+// 'element' is a jquery element for the targeted input field
+wca.applyCurrencyMask = function(action, element, currencyIsoCode) {
+  let entry = getCurrencyInfo(currencyIsoCode);
+  let currentVal = 0;
+
+  // Get current val
+  if (action == "update") {
+    currentVal = getValueInCurrency(element);
+  } else if (action == "init") {
+    currentVal = element.val();
+  } else {
+    throw new Error('Unsupported action for currency mask');
+  }
+
+  // Reconfigure
+  $.data(element[0], "current_subunit_to_unit", entry.subunit_to_unit);
+  let maskOptions = {
+    currencySymbol: entry.symbol,
+    currencySymbolPlacement: (entry.symbol_first) ? 'p' : 's',
+    // If the currency has no subunit (subunit_to_unit is 1), then we don't need
+    // decimals. For currencies with subunits we want to show decimals.
+    decimalPlaces: (entry.subunit_to_unit == 1) ? 0 : 2,
+    showWarnings: false,
+  };
+
+  if (autoNumericObjectMap.has(element)) {
```

Comment:
Interesting, you can use jquery object as keys in a javascript `Map` and there are no issues?

Instead of this `autonumericObjectMap`, could you use jquery's data method instead? https://api.jquery.com/jquery.data/. It looks like it's built to let you store arbitrary data with each object.

Reply:
That's what I understood from [here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map)!

I modified it to use data anyway!